### ISSUE TITLE
How to deal with logits from position embeddings in the output layer?

### ISSUE THREAD

**Issue by xiaoda99** [openai/finetune-transformer-lm#22]
Dear guys,

I found that the position embeddings are concatenated with the word embeddings in the embedding layer. 
https://github.com/openai/finetune-transformer-lm/blob/bd1cf7d678926041e6d19193cab7e5cd8ce2fce6/train.py#L411
and the output layer also shares weights with this embedding layer.
https://github.com/openai/finetune-transformer-lm/blob/bd1cf7d678926041e6d19193cab7e5cd8ce2fce6/train.py#L176

My questions are:
1. During lm pretraining, did you mask out the logits from those position embeddings when computing the loss?
2. If I use the pretrained model as a LM to generate text, do I need to mask out these position embedding logits before softmax when sampling the next word?

BTW, I used the pytorch code ported by huggingface:
https://github.com/huggingface/pytorch-openai-transformer-lm
FYI, I also posted issue there describing some details of my experiments:
https://github.com/huggingface/pytorch-openai-transformer-lm/issues/36

- Da Xiao

**Comment by madisonmay** [comment]
@xiaoda99 in https://github.com/IndicoDataSolutions/finetune/blob/development/finetune/base.py#L544 masking the positional embeddings helped produce more reasonable generated text.
